selfdrive/debug/print_docs_diff.py

```diff
@@ -1,120 +1,93 @@
+    return {k: d[k] for k in d if k in ks}
+  cars_in_common = [(base_cars[name], new_cars[name]) for name in new_cars.keys() if name in base_cars]
+  return [(base, new) for base, new in cars_in_common if base != _limited_to_keys(base.keys(), new)]
 
 
-def format_row(builder):
-  return "|" + "|".join(builder) + "|"
+def find_removed_cars(base_cars, new_cars):
+  return [base_cars[k] for k in base_cars.keys() if k not in new_cars]
 
 
-def print_car_docs_diff(path):
-  base_car_docs = defaultdict(list)
-  new_car_docs = defaultdict(list)
+def find_added_cars(base_cars, new_cars):
+  return [new_cars[k] for k in new_cars.keys() if k not in base_cars]
 
-  for car in load_base_car_docs(path):
-    base_car_docs[car.car_fingerprint].append(car)
-  for car in get_all_car_docs():
-    new_car_docs[car.car_fingerprint].append(car)
 
-  # Add new platforms to base cars so we can detect additions and removals in one pass
-  base_car_docs.update({car: [] for car in new_car_docs if car not in base_car_docs})
+def build_row(headers, base_car, new_car=None):
+  def _preprocess_cell(v):
+    v = re.sub(r'\bassets/', 'https://media.githubusercontent.com/media/commaai/openpilot/master/docs/assets/', v)  # replace 'assets/' with the full URL
+    v = re.sub(r'\[<sup>(.*?)<\/sup>\]\(#footnotes\)', r'<sup>\1</sup>', v)  # remove links to #footnotes
+    return v
+  row_builder = []
+  for header in headers:
+    if new_car and new_car.get(header) != base_car.get(header):
+      row_builder.append(f"{_preprocess_cell(base_car.get(header, ''))} {ARROW_SYMBOL} {_preprocess_cell(new_car.get(header, ''))}")
+    else:
+      row_builder.append(_preprocess_cell(base_car[header]))
+  return "|" + "|".join(row_builder) + "|"
 
-  changes = defaultdict(list)
-  for base_car_model, base_cars in base_car_docs.items():
-    # Match car info changes, and get additions and removals
-    new_cars = new_car_docs[base_car_model]
-    car_changes, car_additions, car_removals = match_cars(base_cars, new_cars)
 
-    # Removals
-    for car_docs in car_removals:
-      changes["removals"].append(format_row([car_docs.get_column(column, STAR_ICON, VIDEO_ICON, FOOTNOTE_TAG) for column in Column]))
+def print_car_docs_diff(new_docs_path):
+  base_docs_content = requests.get("https://raw.githubusercontent.com/commaai/openpilot/master/docs/CARS.md").text
 
-    # Additions
-    for car_docs in car_additions:
-      changes["additions"].append(format_row([car_docs.get_column(column, STAR_ICON, VIDEO_ICON, FOOTNOTE_TAG) for column in Column]))
+  with open(new_docs_path) as new_docs_file:
+    new_docs_content = new_docs_file.read()
 
-    for new_car, base_car in car_changes:
-      # Column changes
-      row_diff = build_column_diff(base_car, new_car)
-      if ARROW_SYMBOL in row_diff:
-        changes["column"].append(row_diff)
+  base_headers, base_cars = get_cars_docs_in_markdown(base_docs_content)
+  new_headers, new_cars = get_cars_docs_in_markdown(new_docs_content)
```

Comment:
It makes more sense if `new_docs_content` is hardcoded using BASEDIR + CARS.md file locally, and `base_docs_content` should be passed in as a path, using the base commit on the PR (downloaded in the Action)

Reply:
I made the change (https://github.com/commaai/openpilot/pull/33273/commits/ce8c77cdb15874ca3eea2555e4830202a58cfa00) and tested it in my fork. Is this what you had in mind?